Test Vector Generation works again

gallizoltan opened the pull request:
bu (bitcoin utils) has been renamed to ku (key utility) and its syntax changed
fixes #34

TechyShishy committed:
Added bip39 seed and bip39 mnemonic fields

In addition to creating a bitcoin address, also generate a mnemonic
key suitable for importing into common HD wallet software.  This
is an advanced function, and not suitable for most users, but may
be useful to power users.

TechyShishy committed:
v1.0.9 Update generated build files for bip39 code

Commit:
makefile use bash instead of sh

make build failed with:
/bin/sh: 1: source: not found

Commit:
README update

Commit:
bu becomes ku

bu renamed to ku (key utility) in pycoin version 0.40

Commit:
pycoin's ku syntax changed

Commit:
bip39 mnemonic for HD wallets #1

gallizoltan committed:
Merge pull request #1 from Shishire/master

bip39 mnemonic for HD wallets

Commit:
README update: install bitcore-mnemonic

Commit:
Merge branch 'mnemonic' of https://github.com/gallizoltan/warpwallet into mnemonic

Commit:
BIP39 Reference Implementation

based on http://github.com/trezor/python-mnemonic
Sep 23, 2013 commit 8dae0cd4c01b219a83b310281febaffa0c13667e

Commit:
bitcore-mnemonic files

files come from npm install bitcore-mnemonic

Commit:
Revert "README update: install bitcore-mnemonic"

This reverts commit 9b7d4225caf9c4d7fd294d698f0a19f949a232a9.

maxtaco commented on README.md:
````diff
@@ -10,7 +10,8 @@ it does use npm to manage its dependencies
 
 ```sh
-$ npm install -g iced-coffee-script
+$ sudo ln -s /usr/bin/nodejs /usr/bin/node
````
Can you remove these changes?  I'm assuming you've installed node as your user and it's in your path.

gallizoltan replied:
Ok, I will remove this.

maxtaco commented on README.md:
````diff
@@ -27,6 +28,8 @@ To generate our reference test vectors, we use the reference Scrypt implementati
 
 ```sh
+$ pip install virtualenv
+$ sudo /usr/bin/easy_install virtualenv
````
is this line needed?

gallizoltan replied:
This is needed only at the first run.

maxtaco commented on test/gen/Makefile:
```diff
@@ -1,8 +1,9 @@
 
+SHELL := /bin/bash
```
why is this needed?  I use `sh`

gallizoltan replied:
Without this the `make build` fails with this message:
```
/bin/sh: 1: source: not found
Makefile:15: recipe for target 'build-python' failed
make: *** [build-python] Error 127
```
I googled for "/bin/sh: 1: source: not found" and found this solution:
https://stackoverflow.com/questions/670191/getting-a-source-not-found-error-when-using-source-in-a-bash-script

I don'n know, I have a Ubuntu 16.04 LTS. What environment are you using for `sh` to find `source`?

maxtaco reviewed (commented):
Thanks for the PR, I'll try it out.

Commit:
mnemonics added to test suite

Regeneration of Test Vectors now also add bip39 mnemonics belongs to the
generated seeds

Commit:
Update to latest BIP39 Reference Implementation

based on http://github.com/trezor/python-mnemonic
Jul 22, 2018 commit 7d23efe34f3dd537996f31dae34f663207dac232

Commit:
using all 32 bytes of seed for mnemonic

Commit:
Regeneration of Test Vectors to include mnemonics

Commit:
mnemonic check included in test

Commit:
bip39-mnemonic field is a textarea

Commit:
version number increase

gallizoltan committed:
Merge pull request #2 from gallizoltan/mnemonic

BIP39 Mnemonic addition

Commit:
Test Vector version number increase

Commit:
generated file update

Commit:
incompatible 1.0.9 generated file removal

Commit:
mention pbkdf2 in README.md

gallizoltan commented:
Sorry for these commits, I was not aware GitHub automatically adds new commits to my pull request. I will close this PR and make a new one, focusing only to make Test Vector Generation work.

gallizoltan commented:
Let's try it again: https://github.com/keybase/warpwallet/pull/36
